docs/running-on-yarn.md

```diff
@@ -270,7 +270,7 @@ To use a custom metrics.properties for the application master and executors, upd
  <td><code>spark.executor.instances</code></td>
   <td><code>2</code></td>
   <td>
-    The number of executors for static allocation. With <code>spark.dynamicAllocation.enabled</code>, the initial set of executors will be at least this large.
+    The number of executors for static allocation. With <code>spark.dynamicAllocation.enabled</code>, the initial set of executors will be at least this large. Default value for static allocation is 2, whereas in dynamic allocation, is 0.
```

Comment:
I think 'the initial set of executors will be at least this large' explains all.

Reply:
@beliefer The default value 2 is not applicable for dynamic allocation case.

That is, if user doesn't set spark.executor.instances, then it takes 0 as the value, in case of dynamic allocation.

Reply:
It actually defaults to the min number of executors. This is already documented on this page, as is the default of 2 here. I wouldn't add this.

Reply:
Yes. In case of dynamic allocation, spark.executor.instances default value is not taken as 2, if user doesn't set. 

So, to clarify this is the documentation I created this PR.

Reply:
This documentation belongs with the 'dynamic allocation' properties, but it's already documented there.

Reply:
Yes. Agree. My point was, the default value is applicable only for static allocation.

Reply:
Intend behind creating the PR is the warning message, when the user gave min executors as 1 and dynamic allocation is enabled.
![spark-jira-executor-msg](https://user-images.githubusercontent.com/23054875/53581783-403e2280-3ba4-11e9-9908-874b9b78f798.png)

Reply:
Looking at the code, it's more complicated. `spark.executor.instances` really has no default. It's taken to be 0 in many parts of the code if not specified. In practice it defaults to 2 from `SchedulerBackendUtils.getInitialTargetExecutorNumber`.

I can see updating the docs to clarify that it has a default of 0, though that would be kind of surprising to find 2 executors then without specifying it. Maybe no default is the most correct answer, but again, 2 is what you end up with in practice.

In any event I don't think it's accurate to say the default changes if you have dynamic allocation enabled. It doesn't apply, really.